arch/arc/core/smp.c

```diff
@@ -130,21 +131,27 @@ static void sched_ipi_handler(const void *unused)
 	z_sched_ipi();
 }
 
-/* arch implementation of sched_ipi */
-void arch_sched_ipi(void)
+void arch_sched_directed_ipi(uint32_t cpu_bitmap)
 {
-	uint32_t i;
+	unsigned int i;
+	unsigned int num_cpus = arch_num_cpus();
 
-	/* broadcast sched_ipi request to other cores
+	/* Send sched_ipi request to other cores
 	 * if the target is current core, hardware will ignore it
 	 */
-	unsigned int num_cpus = arch_num_cpus();
 
 	for (i = 0U; i < num_cpus; i++) {
-		z_arc_connect_ici_generate(i);
+		if ((cpu_bitmap & BIT(i)) != 0) {
+			z_arc_connect_ici_generate(i);
+		}
```

Comment:
Ok, now we have issue here for ARC as core ID and ARConnect ID doesn't always match. Currently we have it so for SMP ARC nSIM (simulation) platforms due to the way we launch it.

I'll check if it is possible to force the IDs so the would match for nSIM platforms as well.

Reply:
Now it's fixed with https://github.com/zephyrproject-rtos/zephyr/pull/72755